bpf/bpf_sock.c

```diff
@@ -1089,8 +1089,10 @@ __sock6_health_fwd(struct bpf_sock_addr *ctx __maybe_unused)
 
 	ctx_get_v6_address(ctx, &addr6);
 	if (is_v4_in_v6(&addr6)) {
+#ifdef ENABLE_IPV4
```

Comment:
I think we need to include also the `is_v4_in_v6()` so that if IPv4 is disabled we do all the lookups in the v6 maps.